e2e/app/modal/base-modal.ts

```diff
@@ -52,14 +53,21 @@ export default abstract class BaseModal extends Container {
       const {waitForNav = false, waitForClose = false, timeout} = waitOptions;
       const button = await this.waitForButton(buttonLabel);
       await button.waitUntilEnabled();
-      await Promise.all(fp.flow(
-         fp.filter<{ shouldWait: boolean, waitFn: () => Promise<void> }>('shouldWait'),
-         fp.map(item => item.waitFn()),
-         fp.concat([button.click()])
-      )([
-         {shouldWait: waitForNav, waitFn: () => this.page.waitForNavigation({waitUntil: ['load', 'networkidle0']})},
-         {shouldWait: waitForClose, waitFn: () => this.waitUntilClose(timeout)}
-      ]));
+      await button.focus();
+      const handle = await button.asElementHandle();
+      await handle.hover();
+      if (waitForNav) {
+         await Promise.all([
+            this.page.waitForNavigation({waitUntil: ['load', 'networkidle0']}),
+            button.click(),
+         ]);
+      } else {
+         await button.click();
+      }
+
+      if (waitForClose) {
+         await this.waitUntilClose(timeout);
+      }
```

Comment:
Reverting code to old-fashioned way to ensure the order of `click` and `waitForNavigation` is kept. `waitForNavigation` happens first before `click` in Promise.all call.